src/core/src/main/java/org/locationtech/geogig/data/EPSGBoundsCalc.java

```diff
@@ -0,0 +1,130 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Given a code string (EPSG:####) or {@link RevFeatureType} , find the CRS bounds and return as an
+ * Envelope
+ */
+public class EPSGBoundsCalc {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Logging.class);
+
+    /**
+     * Get the bounds of the desired CRS, uses JTS ReferencedEnvelope transform to properly
+     * handle polar projections
+     * @param crs the target CoordinateReferenceSystem
+     * @return bounds an Envelope containing the CRS bounds, throws a NoSuchAuthorityCodeException
+     * if the crs cannot be found
+     */
+    public Envelope getExtents(CoordinateReferenceSystem crs) throws Exception {
+
+        final Extent domainOfValidity = crs.getDomainOfValidity();
+
+        if (null == domainOfValidity) {
+            throw new Exception("No domain of validity provided by CRS definition");
+        }
+
+        Collection<? extends GeographicExtent> geographicElements;
+        geographicElements = domainOfValidity.getGeographicElements();
+
+        GeographicExtent geographicExtent = geographicElements.iterator().next();
+        GeographicBoundingBox geographicBoundingBox = (GeographicBoundingBox) geographicExtent;
+
+        double minx = geographicBoundingBox.getWestBoundLongitude();
+        double miny = geographicBoundingBox.getSouthBoundLatitude();
+        double maxx = geographicBoundingBox.getEastBoundLongitude();
+        double maxy = geographicBoundingBox.getNorthBoundLatitude();
+
+        CoordinateReferenceSystem wgs84LongFirst;
+
+        wgs84LongFirst = CRS.decode("EPSG:4326", true);
+        CoordinateOperationFactory coordOpFactory = CRS.getCoordinateOperationFactory(true);
+        CoordinateOperation op = coordOpFactory.createOperation(wgs84LongFirst,crs);
+
+        ReferencedEnvelope refEnvelope = new ReferencedEnvelope(minx, maxx, miny, maxy, wgs84LongFirst);
+        GeneralEnvelope genEnvelope = CRS.transform(op, refEnvelope);
+
+        double xmax = genEnvelope.getMaximum(0);
+        double ymax = genEnvelope.getMaximum(1);
+        double xmin = genEnvelope.getMinimum(0);
+        double ymin = genEnvelope.getMinimum(1);
+
+        Envelope envelope = new Envelope(xmin, xmax, ymin, ymax);
+        return envelope;
+    }
+
+    /**
+     * Search for the given CRS (EPSG code), return the bounds (domain of validity)
+     * @param refId the input CRS
+     * @return projectionBounds an Envelope describing the CRS bounds, throws
+     * NoSuchAuthorityException if the
+     * CRS is not found
+     */
+    public Envelope findCode(String refId) throws Exception {
+        Envelope projectionBounds = null;
+        CoordinateReferenceSystem crs = null;
+
+        CRSAuthorityFactory authorityFactory = CRS.getAuthorityFactory(true);
+
+        crs = authorityFactory.createCoordinateReferenceSystem(refId);
+
+        projectionBounds = getExtents(crs);
+        return projectionBounds;
```

Comment:
see above, just `return getExtents(crs);` and don't bother creating a variable for it.